### PR TITLE
Add simple CI for running the unit tests

### PR DESCRIPTION
**Pull request by f-trivino**
This commit adds a github workflow so that we can keep code quality high with consistent testing, and linting.

The workflow is simple and basically runs lint with Flake8 and the django unit tests.

Signed-off-by: Francisco Trivino <ftrivino@redhat.com>

### CODE REVIEW

**Comment by netoarmando** [comment]
I'd suggest running the "makemigrations" and "migrate" commands too.

File: .github/workflows/gating.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      uses: actions/checkout@v3
+
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r src/install/requirements.txt
+
+    - name: Lint with Flake8
+      run: |
+        flake8 .
```
Comment (flo-renaud) [inline .github/workflows/gating.yaml:27]: Is flake8 installed by default? The yaml file probably needs to add a step installing it with
```
      - name: Install flake8
        run: pip install flake8
```

Comment (f-trivino) [reply .github/workflows/gating.yaml]: thanks @flo-renaud , you are right. Fixed, btw, I had other issues in the workflow, here I have a green run:

https://github.com/f-trivino/ipa-tuura/actions/runs/3251327071/jobs/5336164986

I'm also adding a commit to address flake8 warnings.

**Comment by f-trivino** [comment]
> I'd suggest running the "makemigrations" and "migrate" commands too.

thanks @netoarmando , I need to build this new habit :)

**Review by flo-renaud — COMMENTED** [review]
Hi @f-trivino 
please find inline comments

File: .github/workflows/gating.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        run: |
+          sudo apt install build-essential libdbus-glib-1-dev libgirepository1.0-dev
+          pip install dbus-python
+
+      - name: Install flake8
+        run: pip install flake8
+
+      - name: Lint with Flake8
+        run: |
+          flake8 . --ignore E265,E123,F821 --exclude ./src/ipa-tuura/root/settings.py
```
[the remark this answers is not in the document]
Comment (f-trivino) [reply .github/workflows/gating.yaml]: good catch, I forgot to exclude the 0001_initial.py generated file. I've removed F821.

File: src/ipa-tuura/ipatuura/views.py [hunk 3]
```diff
@@ -2,6 +2,6 @@
 # Copyright (C) 2022  FreeIPA Contributors see COPYING for license
 #
 
-from django.shortcuts import render
+#from django.shortcuts import render
```
[the remark this answers is not in the document]
Comment (f-trivino) [reply src/ipa-tuura/ipatuura/views.py]: done, thanks.

File: src/ipa-tuura/ipatuura/admin.py [hunk 4]
```diff
@@ -2,6 +2,6 @@
 # Copyright (C) 2022  FreeIPA Contributors see COPYING for license
 #
 
-from django.contrib import admin
+#from django.contrib import admin
```
[the remark this answers is not in the document]
Comment (f-trivino) [reply src/ipa-tuura/ipatuura/admin.py]: done, thanks

File: src/ipa-tuura/ipatuura/sssd.py [hunk 5]
```diff
@@ -85,7 +85,7 @@ def __init__(self):
                 DBUS_SSSD_NAME, DBUS_SSSD_GROUPS_PATH)
             self._groups_iface = dbus.Interface(
                 self._groups_obj, DBUS_SSSD_GROUPS_IF)
-        except dbus.DBusException as e:
+        except dbus.DBusException:
```
[the remark this answers is not in the document]
Comment (f-trivino) [reply src/ipa-tuura/ipatuura/sssd.py]: done, thanks

**Comment by f-trivino** [comment]
hi @flo-renaud , green run: https://github.com/f-trivino/ipa-tuura/actions/runs/3272847509/jobs/5384344888

File: src/ipa-tuura/ipatuura/sssd.py [hunk 6]
```diff
@@ -238,14 +238,14 @@ def find_user_groups(self, username):
         """
 
         try:
-            groups = self._sssd_iface.GetUserGroups(name)
+            groups = self._sssd_iface.GetUserGroups(username)
```
Comment (f-trivino) [inline src/ipa-tuura/ipatuura/sssd.py:241]: @flo-renaud , this is supposed to be "username" right?

Comment (flo-renaud) [reply src/ipa-tuura/ipatuura/sssd.py:241]: Yes, good catch.

**Review by flo-renaud — APPROVED** [review]
Thanks for the pr, ack.

**Pull request merged**